File: backend/internal/domain/image.go
package domain

import "github.com/google/uuid"

// Image represents an image entity in the domain
type Image struct {
	ID      uuid.UUID `json:"id"`
	Creator uuid.UUID `json:"creator"`
	Post    Post      `json:"post"`
}

// Post represents a post entity in the domain
type Post struct {
	ID      uuid.UUID `json:"id"`
	Content string    `json:"content"`
}

File: frontend/src/types/image.ts
// 画像関連の型定義（OpenAPI仕様準拠）

export interface Post {
  id: string // UUID
  content: string
}

export interface Image {
  id: string // UUID
  creator: string // UUID
  post: Post
}

export interface ImageDetail {
  id: string // UUID
  creator: string // UUID
  post: Post
}

// 画像検索用のクエリパラメータ
export interface GetImagesParams {
  word?: string // 検索キーワード
}
